types/compat/re.pp

```diff
@@ -0,0 +1,3 @@
+# Emulate the validate_re function
+# validate_re(value, re) translates to Pattern[re], which is not valid puppet and can't be aliased.
```

Comment:
Actually `Pattern[re]` is valid Puppet, but puppet does not (yet) have the ability to create parameterized aliases. Pattern accepts regexp in string form, regexp and (IIRC) a Regexp type.